src/ledgerapp.ts

```diff
@@ -141,7 +141,22 @@ export interface IovLedgerAppSignature extends IovLedgerAppErrorState {
 export function isIovLedgerAppSignature(
   data: IovLedgerAppSignature | IovLedgerAppErrorState,
 ): data is IovLedgerAppSignature {
-  return (data as IovLedgerAppSignature).signature.length > 0;
+  // We could first ensure that there was no error and then proceed to see
+  // if passed object actually as a signature
+  if (data.returnCode === ERROR_CODE.NoError) {
```

Comment:
IMHO, this would be cleaner if the function was short-circuited on condition failure, eg
```
if (data.returnCode !== ERROR_CODE.NoError) return false;

if ("signature" in data && typeof data.signature === "undefined") return false;

const { signature } = data;

return signature.length > 0;
```
With your juicy comments, of course.